cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -105,6 +106,23 @@ runNode
   :: PartialNodeConfiguration
   -> IO ()
 runNode cmdPc = do
+    -- Workaround to ensure that the main thread throws an async exception on
+    -- receiving a SIGTERM signal.
+#ifdef UNIX
```

Comment:
Does this only work on POSIX?

Reply:
Yes, so I've limited scope of the PR to posix, aiming to leave the windows behavior unchanged. I'm honestly not sure what the counterpart is on windows or if one even exits.

Reply:
The primary user of this is system-level benchmarking -- and we don't really run that on Windows, since that platform is wallet-only, @newhoggy.

So we're perfectly find with `#ifdef UNIX`